clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/InMemoryEventRepository.kt

```diff
@@ -0,0 +1,165 @@
+    newEvents.forEachIndexed { index, sagaEvent ->
+      // TODO(rz): Plugin more metadata (provenance, serviceVersion, etc)
+      sagaEvent.metadata = EventMetadata(
+        sequence = currentSequence + (index + 1),
+        originatingVersion = originatingVersion
+      )
+    }
+
+    registry.counter(eventWriteCountId).increment(newEvents.size.toLong())
+    this.events[aggregate]!!.addAll(newEvents)
+    aggregate.version = aggregate.version + 1
+
+    log.debug("Saved $aggregateType/$aggregateId@${aggregate.version}: " +
+      "[${newEvents.joinToString(",") { it.javaClass.simpleName }}]")
+
+    newEvents.forEach { applicationEventPublisher.publishEvent(it) }
+  }
+
+  override fun list(aggregateType: String, aggregateId: String): List<SpinnakerEvent> {
+    registry.counter(eventReadCountId).increment()
+
+    return getAggregate(aggregateType, aggregateId)
+      .let {
+        events[it]?.toList()
+      }
+      ?: throw MissingAggregateEventsException(aggregateType, aggregateId)
+  }
+
+  override fun listAggregates(aggregateType: String?): List<Aggregate> {
+    val aggregates = events.keys
+    return if (aggregateType != null) {
+      aggregates.filter { it.type == aggregateType }
+    } else {
+      aggregates.toList()
+    }
+  }
+
+  private fun getAggregate(aggregateType: String, aggregateId: String): Aggregate {
+    registry.counter(aggregateReadCountId).increment()
+
+    val aggregate = Aggregate(
+      aggregateType,
+      aggregateId,
+      0L
+    )
+    events.putIfAbsent(aggregate, mutableListOf())
+    return events.keys.first { it == aggregate }
+  }
+
+  @Scheduled(fixedDelayString = "\${spinnaker.clouddriver.eventing.memory-repository.cleanup-job-delay-ms:60000}")
+  private fun cleanup() {
+    registry.counter(eventReadCountId).increment()
+
+    config.maxAggregateAgeMs
+      ?.let { Duration.ofMillis(it) }
+      ?.let { maxAge ->
+        val horizon = Instant.now().minus(maxAge)
+        log.info("Cleaning up aggregates last updated earlier than $maxAge ($horizon)")
+        events.entries
+          .filter { it.value.any { event -> event.metadata.timestamp.isBefore(horizon) } }
+          .map { it.key }
+          .forEach {
+            log.trace("Cleaning up $it")
+            events.remove(it)
+          }
+      }
+
+    config.maxAggregatesCount
+      ?.let { maxCount ->
+        log.info("Cleaning up aggregates to max $maxCount items, pruning by earliest updated")
+        events.entries
+          // Flatten into pairs of List<Aggregate, SpinnakerEvent>
+          .flatMap { entry ->
+            entry.value.map { Pair(entry.key, it) }
+          }
+          .sortedBy { it.second.metadata.timestamp }
+          .subList(0, max(events.size - maxCount, 0))
+          .forEach {
+            log.trace("Cleaning up ${it.first}")
+            events.remove(it.first)
+          }
+      }
+  }
+
+  @Scheduled(fixedRate = 1_000)
```

Comment:
every second seems a bit aggressive, no? not sure what the actual usage is (how many events are coming /going).
do you think it's worth making this recurrence interval configurable?